lib/proxy/mutators/urlencoded_bodyparser.js

```diff
@@ -17,58 +17,19 @@ module.exports = function(proxy) {
   return function(req, res, next) {
     // HACK: Remove incompatible headers from the request, and stash them for later.
     proxy.logger.debug(module_tag, "%s %s: headers\n", req.method, req.url, req.headers);
-    // if (typeof req.headers['content-encoding'] !== 'undefined' && req.headers['content-encoding'] !== '') {
-    //   req.original_content_encoding = req.headers['content-encoding'];
-    //   delete req.headers['content-encoding'];
-    //   proxy.logger.warn(module_tag, 'Stashed original "Content-Encoding": "%s"', req.original_content_type);
-    // }
-    // if (typeof req.headers['content-type'] !== 'undefined' && req.headers['content-type'].match(/\s*charset\s*=(?!utf-8)/i)) {
-    //   req.original_content_type = req.headers['content-type'];
-    //   delete req.headers['content-type'];
-    //   req.encoding = req.original_content_type;
-    //   proxy.logger.warn(module_tag, 'Stashed original "Content-Type": "%s"', req.original_content_type);
-    // }
-
-    // If the headers indicate that the content is not UTF-8, then convert the contents to UTF-8.
-    // var test = /^.*?charset\s*=([A-z0-9-_]+)\s*(;|\s)?/i;
-    // var original_encoding = (req.original_content_encoding || req.original_content_type || '').replace(test, '$1');
-    // if (original_encoding !== '' && iconv.encodingExists(original_encoding)) {
-    //   if (req.url !== '') {
-    //     var original_url = req.url;
-    //     req.url = iconv.decode(original_url, original_encoding);
-    //     proxy.logger.debug(module_tag, 'req.url: (%s) %s changed to (UTF-8) %s', original_encoding, util.inspect(original_url), util.inspect(req.url));
-    //   }
-    //   if (typeof req.body !== 'undefined' && req.body !== '') {
-    //     var original_body = req.body;
-    //     req.body = iconv.decode(original_body, original_encoding);
-    //     proxy.logger.debug(module_tag, 'req.body: (%s) %s changed to (UTF-8) %s', original_encoding, util.inspect(original_body), util.inspect(req.body));
-    //   }
-    // } else {
-    //   proxy.logger.error(module_tag, 'Interpreted charset "%s" is invalid', original_encoding);
-    // }
+    if (req.headers['content-type'] && req.headers['content-type'].indexOf('application/x-www-form-urlencoded') === 0) {
+      req.cached_header = req.headers['content-type'];
+      req.headers['content-type'] = 'application/x-www-form-urlencoded';
```

Comment:
LGTM. 

Wow I feel silly for having done so many other unnecessary things. :sweat: I overcomplicated a solution and, evidently, didn't fully understand the problem in the first place. 

Merging this and cleaning up the upstream PR. :+1: